### PR TITLE
Add metric for permanently failed notifications

### PR DESCRIPTION
**Pull request by mneverov**
Fixes: #2361 

Signed-off-by: Max Neverov <neverov.max@gmail.com>

### CODE REVIEW

**Comment by mneverov** [comment]
@SuperQ could you please take a look?

File: notify/notify.go [hunk 1]
```diff
@@ -665,8 +672,9 @@ func (r RetryStage) Exec(ctx context.Context, l log.Logger, alerts ...*types.Ale
 			r.metrics.notificationLatencySeconds.WithLabelValues(r.integration.Name()).Observe(time.Since(now).Seconds())
 			r.metrics.numNotifications.WithLabelValues(r.integration.Name()).Inc()
 			if err != nil {
-				r.metrics.numFailedNotifications.WithLabelValues(r.integration.Name()).Inc()
+				r.metrics.numTotalFailedNotifications.WithLabelValues(r.integration.Name()).Inc()
 				if !retry {
+					r.metrics.numPermanentlyFailedNotifications.WithLabelValues(r.integration.Name()).Inc()
```
Comment (simonpasquier) [inline notify/notify.go:677]: the metric needs to be incremented before L701 too (when the context is done).

Comment (mneverov) [reply notify/notify.go:677]: you're right. Fixed, thanks.

File: notify/notify.go [hunk 2]
```diff
@@ -225,11 +226,16 @@ func newMetrics(r prometheus.Registerer) *metrics {
...
 		}, []string{"integration"}),
-		numFailedNotifications: prometheus.NewCounterVec(prometheus.CounterOpts{
+		numTotalFailedNotifications: prometheus.NewCounterVec(prometheus.CounterOpts{
 			Namespace: "alertmanager",
 			Name:      "notifications_failed_total",
 			Help:      "The total number of failed notifications.",
 		}, []string{"integration"}),
+		numPermanentlyFailedNotifications: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Namespace: "alertmanager",
+			Name:      "notifications_permanently_failed_total",
```
Comment (simonpasquier) [inline notify/notify.go:236]: Not totally convinced by the metric name. We should also count the number of tries.
I'm wondering if we shouldn't use the existing `alertmanager_notifications_failed_total` to count notifications that couldn't be delivered permanently and introduce `alertmanager_notification_requests_total`/`alertmanager_notification_requests_failed_total` metrics to count notification request attempts and failures.

Comment (mneverov) [reply notify/notify.go:236]: @simonpasquier thanks for the update. I tried to be non-intrusive and keep the existing meaning for the `notifications_failed_total`. I agree that your suggested metrics have much clearer intent given these names. 
Should I introduce them and delete the old metric or just add the new ones?

**Comment by mneverov** [comment]
@simonpasquier could you please have another look?

**Review by simonpasquier — COMMENTED** [review]
:+1: thanks!

**Pull request merged**